frontend/lib/src/components/elements/Video/Video.tsx

```diff
@@ -36,12 +38,27 @@ export default function Video({
   element,
   width,
   endpoints,
+  elementMgr,
 }: VideoProps): ReactElement {
   const videoRef = useRef<HTMLVideoElement>(null)
 
+  const [state] = useState<string>(
+    () => elementMgr.getElementState(element.id) || element.id
+  )
```

Comment:
One potential change we are considering for `getElementState` / `setElementState` is to add an additional level via `keys`, e.g.:

```typescript
elementMgr.getElementState(element.id, "rendered")
elementMgr.setElementState(element.id, "rendered", true)
```

This would also make it more obvious in this situation what the state is referring to.

Reply:
@LukasMasuch I used the new implementation with keys. Let me know if using it incorrectly or suboptimally